### PR TITLE
feat(kuma-cp) Prometheus metrics over mTLS

### PR DESCRIPTION
**Pull request by jakubdyszkiewicz**
### Summary

This PR introduces Prometheus metrics over mTLS. There are new properties in Prometheus backend configuration:
```
conf:
  tags:
    service: dataplane-metrics
  skipMTLS: false
```
Since metrics will be exposed with mTLS, there has to be a service that TrafficPermission will match on. You can define this service name in tags. `skipMTLS` lets you skip mTLS defined on the mesh. If the value is nil or true, Prometheus endpoint will be exposed without mTLS.

`kumactl install metrics` now installs prometheus as a part of the mesh with direct access of `"*"`.
I needed to remove probes for now because health checks are not accessible when mTLS is on.
Node Exporter for some reason was crashing the whole K8S cluster when Kuma DP was deployed there, therefore for now sidecar injection for it is disabled.

I introduced the concept of additional networking in the code, so policies can be matched on extra listeners that we are generating automatically (like Prometheus)

### Documentation

TODO (do not merge without PR to docs)

### CODE REVIEW

File: app/kumactl/cmd/install/testdata/install-metrics.defaults.golden.yaml [hunk 1]
```diff
@@ -4847,6 +4849,8 @@ spec:
       labels:
         component: "node-exporter"
         app: prometheus
+      annotations:
+        kuma.io/sidecar-injection: "disabled" # disabled for now, injecting DP crashes K8S cluster
```
Comment (nickolaev) [inline app/kumactl/cmd/install/testdata/install-metrics.defaults.golden.yaml:4853]: Hmm, is this specific to Kind or Minikube has this problem too? Is this something we want to take a second look at?

File: pkg/xds/generator/prometheus_endpoint_generator.go [hunk 2]
```diff
@@ -66,11 +63,40 @@ func (g PrometheusEndpointGenerator) Generate(ctx xds_context.Context, proxy *co
...
+	}
+	iface, err := proxy.Dataplane.Spec.GetNetworking().ToInboundInterface(inbound)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not convert inbound interface")
+	}
+
+	// We assume that skipMTLS = nil is the same as false, so we don't break deployments between 0.5 and 0.5.1
+	skipMTLS := prometheusEndpoint.SkipMTLS == nil || prometheusEndpoint.SkipMTLS.Value
+	var listener *envoy_api.Listener
+	if skipMTLS || !ctx.Mesh.Resource.MTLSEnabled() {
```
Comment (nickolaev) [inline pkg/xds/generator/prometheus_endpoint_generator.go:78]: Would it make sense to handle skipMTLS in a separate function?

File: pkg/xds/generator/prometheus_endpoint_generator_test.go [hunk 3]
```diff
@@ -264,7 +274,7 @@ var _ = Describe("PrometheusEndpointGenerator", func() {
                 trafficDirection: INBOUND
                 address:
                   socketAddress:
-                    address: 0.0.0.0
+                    address: 192.168.0.1
```
Comment (nickolaev) [inline pkg/xds/generator/prometheus_endpoint_generator_test.go:277]: I am probably missing a seirous part of the idea behind this PR, but why we need to change the address here?

File: app/kumactl/cmd/install/testdata/install-metrics.defaults.golden.yaml (hunk 1 above)
Comment (jakubdyszkiewicz) [reply app/kumactl/cmd/install/testdata/install-metrics.defaults.golden.yaml:4853]: It crashed both Minikube and Kind. Maybe it's a matter of DeamonSet

File: pkg/xds/generator/prometheus_endpoint_generator_test.go (hunk 3 above)
Comment (jakubdyszkiewicz) [reply pkg/xds/generator/prometheus_endpoint_generator_test.go]: First of all, it makes sense to expose this on `networking.address`, which is an address that DP is available on publicly. We didn't have this primitive back then when we were building metrics.
Second of all, `0.0.0.0` is potentially more conflicts. DirectAccessProxyGenerator would be more complicated.

File: pkg/xds/generator/prometheus_endpoint_generator.go (hunk 2 above)
Comment (jakubdyszkiewicz) [reply pkg/xds/generator/prometheus_endpoint_generator.go]: Refactor

File: app/kumactl/cmd/install/testdata/install-metrics.defaults.golden.yaml (hunk 1 above)
Comment (nickolaev) [reply app/kumactl/cmd/install/testdata/install-metrics.defaults.golden.yaml:4853]: Please file an issue with the information you have.

**Comment by jakubdyszkiewicz** [comment]
Issue created. Docs are here https://github.com/Kong/kuma-website/pull/214

**Pull request merged**